### ISSUE TITLE
NextjsSite: setting defaultFunctionProps timeout causes undefined function name

### ISSUE THREAD

**Issue by nicolas-lescop** [serverless-stack/sst#1179]
Hello,

When I try to change lambda timeout it throw this cloud formation error and my stack is stuck in the **UPDATE_ROLLBACK_FAILED** state.

Cloud formation error:
```
Received response status [FAILED] from custom resource. Message returned: Missing required key 'FunctionName' in params
```

Error in cloud watch:
```
INFO	[provider-framework] MissingRequiredParameter: Missing required key 'FunctionName' in params
    at ParamValidator.fail (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:50:37)
    at ParamValidator.validateStructure (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:61:14)
    at ParamValidator.validateMember (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:88:21)
    at ParamValidator.validate (/var/runtime/node_modules/aws-sdk/lib/param_validator.js:34:10)
    at Request.VALIDATE_PARAMETERS (/var/runtime/node_modules/aws-sdk/lib/event_listeners.js:132:42)
    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at callNextListener (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at /var/runtime/node_modules/aws-sdk/lib/event_listeners.js:86:9
    at finish (/var/runtime/node_modules/aws-sdk/lib/config.js:386:7)
    at /var/runtime/node_modules/aws-sdk/lib/config.js:404:9 {
  code: 'MissingRequiredParameter',
  time: 2021-12-23T23:03:29.829Z
}
```

Config to repro:
```
const site = new NextjsSite(this, 'NextSite', {
      path: 'apps/next',
      defaultFunctionProps: {
        timeout: 20,
      },
});
```

My stack is way more complex but this error happens when I try to set **timeout**

Versions:
```
"aws-sdk": "^2.817.0",
"@serverless-stack/cli": "0.54.4",
"@serverless-stack/resources": "0.54.4",
"@serverless-stack/static-site-env": "0.54.4",
```

Thanks

**Comment by elliehamilton3** [comment]
@nicolas-lescop Did you ever get a solution for this issue? We have the same problem

**Comment by fwang** [comment]
This is fixed in v1.6.7.

**Comment by fwang** [comment]
@nicolas-lescop sorry about the delay in getting to this one 🙏
